core/src/main/java/ai/timefold/solver/core/impl/move/streams/MoveIterable.java

```diff
@@ -0,0 +1,12 @@
+package ai.timefold.solver.core.impl.move.streams;
+
+import java.util.Iterator;
+import java.util.Random;
+
+import ai.timefold.solver.core.preview.api.move.Move;
+
+public interface MoveIterable<Solution_> extends Iterable<Move<Solution_>> {
```

Comment:
I'm not sure what the advantage is of creating this abstraction

Reply:
We need to support two kinds of iteration - original (plain `Iterable`) and random (`iterator(Random)`). Therefore this abstraction, which exposes both of these methods.